numpy/core/src/multiarray/arrayobject.c

```diff
@@ -1269,6 +1270,26 @@ DEPRECATE_silence_error(const char *msg) {
     return 0;
 }
 
+/*
+ * Silence the current error and emit a deprecation future warning instead.
+ *
+ * If warnings are raised as errors, this sets the warning __cause__ to the
+ * silenced error.
+ */
+NPY_NO_EXPORT int
+DEPRECATE_FUTUREWARNING_silence_error(const char *msg) {
```

Comment:
Maybe better to just add a `PyObject *warning_type` argument to the original function, and use `PyErr_WarnEx(warning_type, msg, 1)` instead.

Reply:
Since I needed the fetch the error anyway, the need for the new function disappeared...